src/Storages/ColumnsDescription.cpp

```diff
@@ -331,8 +331,7 @@ NamesAndTypesList ColumnsDescription::getAll() const
 
 bool ColumnsDescription::has(const String & column_name) const
 {
-    return columns.get<1>().find(column_name) != columns.get<1>().end()
-        || subcolumns.find(column_name) != subcolumns.end();
+    return columns.get<1>().find(column_name) != columns.get<1>().end();
```

Comment:
How `There is no column ... in table` is possible if previous condition is less strict than current? That is, it seems that if column was not found with the previous condition, then it won't be found with the current one.

Reply:
The problem was, that `has` and `get` were inconsistent. Previously `has` returned true for subcolumns, while `get` throwed exception. And the following code was broken ``` if (columns.has(name)) columns.get(name)```